osu.Game/Screens/Select/SongSelect.cs

```diff
@@ -574,6 +574,8 @@ protected virtual void UpdateBeatmap(WorkingBeatmap beatmap)
                 beatmap.Track.Looping = true;
         }
 
+        public virtual void ShowResults() => BeatmapDetails.Leaderboard.ScoreSelected += s => this.Push(new SoloResults(s));
```

Comment:
Keep it as it was before, but add/use a `CreateResultsScreen(ScoreInfo score)` method.